php-checks/src/main/java/org/sonar/php/checks/WeakSSLProtocolCheck.java

```diff
@@ -106,39 +119,90 @@ public void visitFunctionCall(FunctionCallTree tree) {
+    var enclosingBlock = TreeUtils.findAncestorWithKind(expressionTree, Tree.Kind.BLOCK);
+    if (enclosingBlock != null) {
+      var usages = new ReadWriteUsages(enclosingBlock, context().symbolTable());
+      var hasSensitiveReassignment = checkOptionsReassignment(usages, expressionTree);
+      if (hasSensitiveReassignment) {
+        // sensitive reassignment detected, no need to check the initial assignment
+        return;
+      }
+    }
+
     ExpressionTree config = getAssignedValue(expressionTree);
     if (!isArrayInitializer(config)) {
       return;
     }
     getProperty((ArrayInitializerTree) config, "SSL")
       .flatMap(sslConfig -> {
         if (isArrayInitializer(sslConfig)) {
-          return getProperty((ArrayInitializerTree) sslConfig, "crypto_method");
+          return getProperty((ArrayInitializerTree) sslConfig, CRYPTO_METHOD_KEY);
         }
         return Optional.empty();
       })
       .ifPresent(value -> checkStreamWeakProtocol(value, STREAM_CONTEXT_CREATE));
   }
 
+  private boolean checkOptionsReassignment(ReadWriteUsages usages, ExpressionTree expressionTree) {
+    var symbol = context().symbolTable().getSymbol(expressionTree);
+    if (symbol == null) {
+      return false;
+    }
+
+    var optionsSslAccesses = usages.getWritesSorted(symbol).stream()
+      .map(SyntaxToken::getParent)
+      .filter(tree -> isArrayObjectWithOffset(tree, "ssl"))
+      .map(Tree::getParent)
+      .toList();
+
+    // In a sensitive case, `$options['ssl'] is either assigned with an array initializer or a value is directly written by an array key
+    // check if `$options['ssl']` is assigned with a sensitive array initializer
+    boolean hasOptionsSslSensitiveAssignment = optionsSslAccesses.stream()
+      .map(WeakSSLProtocolCheck::getAssignedValueFromParent)
+      .filter(param -> param != null && isArrayInitializer(param))
+      .findFirst()
+      .flatMap(c -> getProperty((ArrayInitializerTree) c, CRYPTO_METHOD_KEY))
+      .map(tree -> checkStreamWeakProtocol(tree, STREAM_CONTEXT_CREATE))
+      .orElse(false);
+
+    if (hasOptionsSslSensitiveAssignment) {
+      return true;
+    }
+
+    // check if `$options['ssl']['crypto_method']` is set directly
+    return optionsSslAccesses.stream()
+      .filter(tree -> isArrayObjectWithOffset(tree, CRYPTO_METHOD_KEY))
+      .map(Tree::getParent)
+      .map(WeakSSLProtocolCheck::getAssignedValueFromParent)
+      .filter(Objects::nonNull)
+      .findFirst()
+      .map(tree -> checkStreamWeakProtocol(tree, STREAM_CONTEXT_CREATE))
+      .orElse(false);
+  }
+
   private static boolean isArrayInitializer(ExpressionTree param) {
     return param.is(Tree.Kind.ARRAY_INITIALIZER_BRACKET, Tree.Kind.ARRAY_INITIALIZER_FUNCTION);
   }
 
-  private void checkStreamWeakProtocol(ExpressionTree expressionTree, String functionName) {
+  private boolean checkStreamWeakProtocol(ExpressionTree expressionTree, String functionName) {
```

Comment:
I added this to control the number of issues we raise. I.e., if there was a reassignment to a sensitive value, there is no need to raise additionally on the initializer, even if it is also sensitive. However, it may be an unnecessary complication for a quite rare case.